File: project_euler/Problem039/Problem039.c
#include <stdio.h>

void Problem039(void) {
    int target = 1000;
    int result = 0;
    int pivot = 0;
    for (int i = target; i >= 2; i -= 2) {
        int count = 0;
        for (int j = i / 3; j > 0; --j) {
            if (!((((i * i) >> 1) - i * j) % (i - j)) && j < (((i * i) >> 1) - i * j) / (i - j)) {
                ++count;
            }
        }
        if (pivot < count) {
            pivot = count;
            result = i;
        }
    }
    printf("%d\n", result);
}

File: project_euler/Problem039/Problem039.py
class Problem039:
    def solution(self) -> None:
        target, result, pivot = 1000, 0, 0
        for i in range(target, 1, -1):
            count = 0
            for j in range(i // 3, 0, -1):
                if not (((i * i) >> 1) - i * j) % (i - j) and j < (((i * i) >> 1) - i * j) // (i - j):
                    count += 1
            if pivot < count:
                pivot, result = count, i
        print(result)
